File: Ejercicio4/pwCapsFilter.rb
require_relative 'filter'

class PwCapsFilter
  include Filter

  def ejecutar(credenciales)
    tiene_mayus = credenciales.password.match?(/[A-Z]/)
    tiene_minus = credenciales.password.match?(/[a-z]/)

    if tiene_mayus and tiene_minus
      puts "Comprobación de minúsculas y mayúsculas cumplida con éxito"
      return true
    else
      puts "Contraseña no válida: debe contener al menos una letra mínuscula
      y otra mayúscula."
    end
  end
end

File: P2/Ejercicio1/lib/pwCapsFilter.dart
import 'filter.dart';
import 'struct_credenciales.dart';

class PwCapsFilter implements Filter {
  static final _regexMayus = RegExp(r'[A-Z]');
  static final _regexMinus = RegExp(r'[a-z]');

  @override
  bool ejecutar(Credenciales credenciales) {
    final tieneMayus = _regexMayus.hasMatch(credenciales.password);
    final tieneMinuscula = _regexMinus.hasMatch(credenciales.password);

    return (tieneMayus && tieneMinuscula);
  }

  @override
  String msgerror(){
    return "Contraseña no válida: debe contener al menos una letra minúscula y otra mayúscula.";
  }
}